content/testing-yaml/testing.md

```diff
@@ -21,39 +23,71 @@ npm test
+  test_report: app/build/test-results/**/*.xml
 ```
 
 UI tests (also known as instrumented tests):
 
 ```bash
-./gradlew connectedAndroidTest
+- name: Launch emulator
+  script: |
+    cd $ANDROID_HOME/tools
+    emulator -avd emulator &
+    adb wait-for-device
+- name: Test
+  script: |
+    set -e
+    ./gradlew connectedAndroidTest
+    adb logcat -d > emulator.log
+  test_report: app/build/outputs/androidTest-results/connected/*.xml
 ```
 
-## Native iOS
+**Tip:**: you can save the emulator log with `adb logcat -d > emulator.log` command.
+
+Instrumentation test with Firbase test lab
 
 ```bash
-set -o pipefail
-xcodebuild \
-    -workspace MyAwesomeApp.xcworkspace \
-    -scheme MyAwesomeApp \
-    -sdk iphonesimulator \
-    -destination 'platform=iOS Simulator,name=iPhone 6,OS=8.1' \
-    test | xcpretty
+    environment:
+      vars:
+        ...
+        GCLOUD_SERVICE_CREDENTIALS: # encrypted key of your google service account with role Editor
+        GCLOUD_PROJECT: # name of your gcloud project
+    scripts:
+      ...
+      - name: Authorize gcloud service account
+        script: |
+          GCLOUD_SERVICE_CREDENTIALS_PATH=$FCI_BUILD_DIR/google-services-credentials.json
+          echo $GCLOUD_SERVICE_CREDENTIALS | base64 --decode > $GCLOUD_SERVICE_CREDENTIALS_PATH
+          gcloud auth activate-service-account --key-file=$GCLOUD_SERVICE_CREDENTIALS_PATH
+          gcloud config set project $GCLOUD_PROJECT
+      - name: Firebase connected test
+        script: |
+          ./gradlew assembleDevDebugAndroidTest                                            # assemble a test to run it on firebase
+          gcloud firebase test android run \
+            --app "app/dev/release/app.aab" \                                              # path to your actual application
+            --test "$(find app/build/outputs/apk/androidTest -name "*.apk" | head -1)" \   # path to the assembled test
+            --device "model=Nexus5X,version=23" \                                          # a device available from gcloud firebase test android models list
+            --timeout "45m" \
+            --type "instrumentation" \
+            --quiet
 ```
 
-If may want to export the test log, you can do this by splitting the standard output to a file
+## Native iOS
 
 ```bash
-set -o pipefail
-xcodebuild \
-    -workspace MyAwesomeApp.xcworkspace \
-    -scheme MyAwesomeApp \
-    -sdk iphonesimulator \
-    -destination 'platform=iOS Simulator,name=iPhone 6,OS=8.1' \
-    test | xcpretty |& tee "/tmp/xcodetest.log"
+- name: iOS test
+    script: |
+    xcode-project run-tests \
+        --workspace MyAwesomeApp.xcworkspace \
+        --scheme MyAwesomeApp \
+        --device "iPhone 11"
+    test_report: build/ios/test/*.xml
 ```
 
+Please check [Codemagic CLI tools documentation](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/xcode-project/run-tests.md#run-tests) to learn about more optional arguments to `xcode-project run-tests`.
```

Comment:
.. to learn more about optional arguments to...

Reply:
fixed in e0d9c8e6222cf49670671fc10c0b345d6644767c